nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-avro-record-utils/src/test/java/org/apache/nifi/avro/TestAvroTypeUtil.java

```diff
@@ -239,4 +243,20 @@ public void testComplicatedRecursiveSchema() {
         Assert.assertEquals(recordASchema, ((RecordDataType)recordBParentField.get().getDataType()).getChildSchema());
     }
 
+    @Test
+    public void testMapWithNullSchema() throws IOException {
+
+        Schema recursiveSchema = new Schema.Parser().parse(getClass().getResourceAsStream("schema.json"));
+
+        // Make sure the following doesn't throw an exception
+        RecordSchema recordASchema = AvroTypeUtil.createSchema(recursiveSchema.getTypes().get(0));
+
+        // check the fix with the proper file
+        try(DataFileStream<GenericRecord> r = new DataFileStream<>(getClass().getResourceAsStream("data.avro"),
```

Comment:
Can you explain more about what's going on here, including what is in the data.avro file? When I run avro-tools tojson on it, I get the following:

```
java -jar avro-tools-1.8.1.jar tojson datasets/data.avro
{"a.A":{"o":{"a.O":{"hash":{"map":{}}}}}}
```

Perhaps it would be good to have a test file that has a record with a non-null value for hash, as well as a record with a null value for hash?

Reply:
Also it could check the records to make sure the field values are what you expect (null vs not-null, e.g.)

Reply:
my uderstanding was that the map presence cause the issue in union type,
 whether or not, the map is filled, (instance of map will be returned)

Reply:
I have the following schema in my test flow: 
`{
 "type": "record",
 "name": "A","fields": [
   {"name": "a", "type": "string"},
   {"name": "c", "type": [ "null", {"type" : "map","values" : "string"} ] } 
 ]
}`

When I debug through the processor to AvroTypeUtil, I get a MapRecord as the type of "value", not Map. If it could be either, we could just check for either?

Reply:
I can post my flow and sample Avro file if you'd like to see what I mean

Reply:
OK let's change that to MapRecord, i'm sure it's reliable, because we use it on dev plateform, for the "instanceof Map" you suggest i'm not sure of the impact.

Reply:
and if you also tests it , it's OK